### DISCUSSION TITLE
How to excute fetch data just in generate?

### DISCUSSION THREAD

**Discussion by bimohxh** [nuxt/nuxt#17105] [Questions]
I have some static article json file in project folder. and I want to generate article page use this json file.  I wraite the following code in articlce vue component

```js
<script setup>
const route = useRoute();
let article = ref(null);

const { data } = await useFetch(`/api/article `, {
    params: {
        name: route.params.name
    }
});

resource = data;
</script>
```
I use nuxt server host the json files. When I run generate, `/api/article`  be requested once, and when I deploy the static files in static server and visite the static site. `/api/article`  be requested again.

But I just want the page to be full static, not run request data at client. so what shoud I do?

**Comment by bimohxh** [comment]
It seems that,  the data will be coverd in client request, so what the generate has do?

**Comment by Flanker72** [comment]
You need some kind of a builder script.
You may create a separate NodeJS script and add it to your package.json. This script should fetch the desred data and generate json files.
Then before the generation you should trigger this script to fetch data and place it in the right place.
And in the Vue component you should use plain generated JSON instead of the API request.
The given solution will ALWAYS fetch data, because it is not a static page, it's a dynamic.

The possible solution is like this.

Structure:
```
/pages
/pages/my-page.vue
/data
/data/my-data.json    // generated by builder script
```

my-page.vue
```ts
import resource from "~~/data/my-data.json"; // It will be bundled in static page
```

**Reply by bimohxh** [reply]
> replies

I have degrade to nuxt2. nuxt2 support this feature.
